Marlin/src/feature/bltouch.h

```diff
@@ -76,7 +77,7 @@ class BLTouch {
     static constexpr bool high_speed_mode = false;
   #endif
 
-  static float z_extra_clearance() { return high_speed_mode ? 7 : 0; }
+  static float z_extra_clearance() { return high_speed_mode ? -probe.offset.z : 0; }
```

Comment:
The extra 7mm is to account for the length of the BLTouch pin (extending below the point where it triggers). The point where the probe triggers is at the exact height defined by `probe.offset.z`.

In "high speed mode" the pin is expected to automatically re-extend as soon as the probe is triggered, and so extra clearance is needed compared to the case where the probe pin remains in the unextended position.

So this change is inappropriate. The value of `probe.offset.z` may be extremely small, as little as 1-2mm, and does not account for the extra length of the extended probe pin.

Reply:
So in reality it needs to be slightly larger than `-probe.offset.z`. 7mm is far too high, and kind of defeats the purpose of HS mode. Surely the extra distance required should actually come from `Z_CLEARANCE_AFTER_PROBING`/ `Z_CLEARANCE_MULTI_PROBE`.

Have you tested my version? works a treat

Reply:
Perhaps a define for the `PROBE_TRIGGER_DISTANCE`?

It would be the distance between the probe touching the bed and the probe triggering, could be added to the -probe offset above.